.travis.yml

```diff
@@ -128,13 +137,40 @@ before_install:
             docker-compose stop galaxy-slurm
         fi
         
-        if [ "${COMPOSE_SLURM}" ]
+        if [ "${COMPOSE_SLURM}" ] || [ "${COMPOSE_SLURM_SINGULARITY}" ]
         then
             # turn down the htcondor services
             docker-compose stop galaxy-htcondor galaxy-htcondor-executor galaxy-htcondor-executor-big
         fi
 
         sleep 30
+        
+        if [ "${COMPOSE_SLURM_SINGULARITY}" ]
+        then
+            # docker-compose is already started and has pre-populated the /export dir
+            # we now turn it down again and copy in an example tool with tool_conf.xml and 
+            # a test singularity image. If we copy this from the beginning, the magic Docker Galax startup
+            # script will not work as it detects something in /export/
+            docker-compose down
+            sudo mkdir -p /export/database/container_images/singularity/mulled/
+            sudo curl -L -o /export/database/container_images/singularity/mulled/samtools:1.4.1--0 https://github.com/bgruening/singularity-galaxy-tests/raw/master/samtools:1.4.1--0
+            sudo curl -L -o /export/cat_tool_conf.xml https://github.com/bgruening/singularity-galaxy-tests/raw/master/cat_tool_conf.xml
+            sudo curl -L -o /export/cat.xml https://github.com/bgruening/singularity-galaxy-tests/raw/master/cat.xml
+
+            docker-compose up -d
+            sleep 60
+
+            parsec init --api_key admin --url localhost:8080
+            HISTORY_ID=$(parsec histories create_history | jq .id -r)
+            DATASET_ID=$(parsec tools paste_content 'asdf' $HISTORY_ID | jq '.outputs[0].id' -r)
+            OUTPUT_ID=$(parsec tools run_tool $HISTORY_ID cat '{"input1": {"src": "hda", "id": "'$DATASET_ID'"}}' | jq '.outputs | .[0].id' -r)
+            sleep 15
+            # we are lucky that the out test image has a small bug and always prints a warning. We use this to grep if the tool was running with SINGULARITY
```

Comment:
There is an option `--quiet` that should help but does not. So I consider this as an bug. More general Singularity should not assume such paths. I'm in contact with upstream about all of this, including the rw,ro bug and a static binary.